src/ToricVarieties/NormalToricVarieties/methods.jl

```diff
@@ -5,35 +5,27 @@
 @doc Markdown.doc"""
     betti_number(v::AbstractNormalToricVariety, i::Int)
 
-Compute the i-th Betti number of the normal toric variety `v`.
+Compute the `i`-th Betti number of the normal toric variety `v`.
 """
 function betti_number(v::AbstractNormalToricVariety, i::Int)
     # check input
-    if i > 2*dim(v) || i < 0
-        return 0
+    d = dim(v)::Int
+    if i > 2*d || i < 0 || isodd(i)
```

Comment:
I like that the method now immediately returns 0 when `i` is odd. As you said - reduces the amount of memory used for storage and circumvents a lot of unnecessary computations. Great!